test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -196,15 +202,19 @@ private void HandleHttpRequests()
                 try
                 {
                     var ctx = _listener.GetContext();
-                    var rawSpans = MessagePackSerializer.Deserialize<dynamic>(ctx.Request.InputStream);
-                    var spans = ToSpans(rawSpans);
 
-                    lock (this)
+                    if (ShouldSerializeTraces)
```

Comment:
Most of our smoke tests (which are our longest running tests) don't actually inspect traces, so this should speed up some of the heavier hitters by quite a bit.

Reply:
```suggestion
                    if (ShouldDeserializeTraces )
```